tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,354 @@
+
+
+def get_program_info(duthost, container_name, program_name):
+    """
+    @summary: Get program running status and its pid by analyzing the command
+              output of "docker exec <container_name> supervisorctl status"
+    @return:  Program running status and its pid
+    """
+    program_status = None
+    program_pid = -1
+
+    program_list = duthost.shell("docker exec {} supervisorctl status".format(container_name))
+    for program_info in program_list["stdout_lines"]:
+        if program_info.find(program_name) != -1:
+            program_status = program_info.split()[1].strip()
+            if program_status == "RUNNING":
+                program_pid = int(program_info.split()[3].strip(','))
+            break
+
+    if program_pid != -1:
+        logger.info("Found program {} in the {} state with pid {}"
+                    .format(program_name, program_status, program_pid))
+
+    return program_status, program_pid
+
+
+def get_container_autorestart_states(duthost):
+    """
+    @summary: Get container names and their autorestart states by analyzing
+              the command output of "show container feature autorestart"
+    @return:  A dictionary where keys are the names of containers which have the
+              autorestart feature implemented and values are the autorestart feature
+              state for that container
+    """
+    container_autorestart_states = {}
+
+    show_cmd_output = duthost.shell(CMD_CONTAINER_FEATURE_AUTORESTART)
+    for line in show_cmd_output["stdout_lines"]:
+        container_name = line.split()[0].strip()
+        container_state = line.split()[1].strip()
+        if container_state in ["enabled", "disabled"]:
+            container_autorestart_states[container_name] = container_state
+
+    return container_autorestart_states
+
+
+def get_disabled_container_list(duthost):
+    """
+    @summary: Get the container/service names which are disabled by default
+    @return: A list includes the names of disabled containers/services
+    """
+    disabled_containers = []
+
+    show_cmd_output = duthost.shell(CMD_FEATURE)
+    for line in show_cmd_output["stdout_lines"]:
+        container_name = line.split()[0].strip()
+        container_state = line.split()[1].strip()
+        if container_state == "disabled":
+            disabled_containers.append(container_name)
+
+    return disabled_containers
+
+
+def is_container_running(duthost, container_name):
+    """
+    @summary: Decide whether the container is running or not
+    @return:  Boolean value. True represents the container is running
+    """
+    is_running = duthost.shell("docker inspect -f \{{\{{.State.Running\}}\}} {}".format(container_name))
+    if is_running["stdout_lines"][0].strip('\n') == "true":
+        return True
+    return False
+
+
+def check_container_state(duthost, container_name, should_be_running):
+    """
+    @summary: Determine whether a container should be in running state or not
```

Comment:
Reword: `Determine whether a container is in the expected state (running/not running)`

Reply:
Updated!